--- a/infobash
+++ b/infobash
@@ -62,17 +62,17 @@ IRCCL=" $IRCCL    \x0312   \x0302   \x03
 COLORS=(DGREY BLACK RED DRED GREEN DGREEN YELLOW DYELLOW BLUE DBLUE MAGENTA DMAGENTA CYAN DCYAN WHITE GREY NORMAL)
 # See above for notes on EMPTY
 SCHEMES=(EMPTY,EMPTY,EMPTY NORMAL,NORMAL,NORMAL BLUE,NORMAL,NORMAL GREEN,YELLOW,NORMAL DYELLOW,NORMAL,NORMAL CYAN,BLUE,NORMAL RED,NORMAL,NORMAL GREEN,NORMAL,NORMAL YELLOW,NORMAL,NORMAL GREEN,DGREEN,NORMAL BLUE,RED,NORMAL BLUE,NORMAL,RED YELLOW,WHITE,GREEN BLUE,NORMAL,GREEN DCYAN,NORMAL,DMAGENTA)
-DEFSCHEME=2	# Defaults to 2, make this 1 for normal, 0 for no colorcodes at all. Set to any other valid scheme you like. Same as runtime parameter.
+DEFSCHEME=0	# Defaults to 2, make this 1 for normal, 0 for no colorcodes at all. Set to any other valid scheme you like. Same as runtime parameter.
 SHOWHOST=1	# Set this to 0 to avoid printing the hostname 
-SHOWIRC=2	# SHOWIRC=1 to avoid showing the irc client version number, or SHOWIRC=0 to disable client information completely.
+SHOWIRC=0	# SHOWIRC=1 to avoid showing the irc client version number, or SHOWIRC=0 to disable client information completely.
 INDENT=14	# Default indentation level
-VBL=0		# Verbosity level defaults to 0, this can also be set with -v, -v2, -v3, etc as a parameter.
+VBL=0		# Verbosity level defaults to 0, this can also be set with -v, -v2, -v3, -v4, etc as a parameter.
 IBSHELL=0	# Running in a shell? Defaults to false, and is determined later.
 NOCPUCOUNT=0	# Wether or not the string "dual" or similar is found in cpuinfo output. If so, avoid dups.
 # In cases of derived distros where the version file of the base distro can also be found under /etc, the derived distro's
 # version file should go first. (Such as with Sabayon / Gentoo)
 DIST_DERIV="aptosid-version sidux-version kanotix-version knoppix-version sabayon-release redhat-release turbolinux-release zenwalk-version"
-DIST_MAIN="gentoo-release mandrake-release slackware-version redhat-release SuSE-release"
+DIST_MAIN="gentoo-release mandrake-release slackware-version redhat-release SuSE-release openwrt_release"
 # debian_version excluded from DIST_MAIN so Debian can fall through to /etc/issue detection. Same goes for Ubuntu.
 DIST_BLIST="debian_version ubuntu_version"
 FL1='-'		# These two determine seperators in single line output, to force irc clients not to break off sections
@@ -133,7 +133,7 @@ checkdepend () {
 	fi
 	
 	# bc removed from deps for now
-	for I in df free gawk grep hostname lspci ps readlink runlevel tr uname uptime wc; do
+	for I in df free gawk grep lspci ps readlink tr uname uptime wc; do
 		type -p $I >/dev/null || error 5 "$I"
 	done
 }
@@ -336,7 +336,7 @@ getparams () {
 					VBL="$1" 
 				fi
 			fi
-			grep -q '^[1-3]$' <<< "$VBL" || error 4 "$VBL"
+			grep -q '^[1-4]$' <<< "$VBL" || error 4 "$VBL"
 		;;
 		--version)
 			say "InfoBash, the universal, portable, system info script for irc."
@@ -411,7 +411,7 @@ getparams "$@"
 # If no colorscheme was set in the parameter handling routine, then set the default scheme
 ((CSCHEME)) || cscheme "$DEFSCHEME"
 
-HN=$(hostname)
+HN=$(uci get system.@system[0].hostname)
 
 unset DISTRO
 cd /etc; DISTGLOB=(*[-_]{release,version}); cd "$OLDPWD"
@@ -436,6 +436,9 @@ else
 	DISTRO=$(gawk '{ gsub(/\\[a-z]/,""); gsub(/ [ ]+/," "); gsub(/^ +| +$/,""); print ; exit }' "/etc/${DISTFILE}")
 fi
 ((${#DISTRO} > 80 && ! CRAP)) && DISTRO="${RED}/etc/${DISTFILE} corrupted, use --crap to override${NORMAL}"
+if [ -r /etc/openwrt_release ]; then
+	DISTRO="$(awk -F\' '/^DISTRIB_DESCRIPTION/{print $2}' /etc/openwrt_release)"
+fi
 : ${DISTRO:=Unknown distro o_O}
 
 OSKERN=$(uname -a | gawk '{print $1,$3,$(NF-1)}')
@@ -630,7 +633,7 @@ END {
 	printf("%.1f/%.1fMB\n", used/1024, tot/1024)
 }' /proc/meminfo)
 
-PROC="$(($(ps aux | wc -l)-1))"
+PROC="$(($(ps | wc -l)-1))"
 UPT=$(uptime | gawk '{ a = gensub(/^.*up *([^,]*).*$/,"\\1","g",$0); gsub(/ /,"",a); print a }')
 
 IFS=$'\n'; GFX=($(gawk -F': ' '/VGA compatible controller/ { print $NF }' <<< "$LSPCI")); IFS="$OIFS"
@@ -739,6 +742,11 @@ if ((VBL)); then  # (IF VBL > 1)
 	LNINTRO="$LNINTRO$(echo -ne "${C2}$OSKERN $FIRMWARE${CN}[ ${C1}$DISTRO ${CN}]")"
 	say "$LNINTRO"
 
+	if [[ $VBL -ge 4 ]]; then
+		LNSYSTEM=$(printf "${C1}%-${INDENT}s${C2} %s" "System" "$(cat /tmp/sysinfo/model)")
+		say "$LNSYSTEM"
+	fi # (IF VBL >= 4)
+
 	##say "CPU[0]=\"${CPU[0]}\""
 	IFS=","; CPUTMP=(${CPU[0]}); IFS="$OIFS" 
 	# Strange (and also some expected) behavior encountered.
